File: TradeItIosEmsApi/TradeItAuthenticationRequest.m
#import "TradeItAuthenticationRequest.h"

@implementation TradeItAuthenticationRequest

-(id) initWithUserToken:(NSString *) userToken userId:(NSString *) userId andApiKey:(NSString *) apiKey andAdvertisingId:(NSString *) advertisingId {
    self = [super init];
    if (self) {
        self.userToken = userToken;
        self.userId = userId;
        self.apiKey = apiKey;
        self.advertisingId = advertisingId;
    }
    
    return self;
}

@end

File: TradeItIosEmsApi/TradeItAuthenticationRequest.swift
class TradeItAuthenticationRequest: TradeItRequest, Codable {
    
    var userToken: String
    var userId: String
    var apiKey: String
    var advertisingId: String?
    
    init(userToken: String, userId: String, apiKey: String, advertisingId: String?) {
        self.userToken = userToken
        self.userId = userId
        self.apiKey = apiKey
        self.advertisingId = advertisingId
    }
}
